src/components/ListItem.jsx

```diff
@@ -1,9 +1,48 @@
 import './ListItem.css';
+import { useEffect, useCallback } from 'react';
+import { updateItem } from '../api/firebase';
+
+const millisecondsIn24hrs = 86400000;
+
+export function ListItem({ listToken, item }) {
+	useEffect(() => {
+		let calcDate = item.dateLastPurchased
+			? item.dateLastPurchased.seconds * 1000
+			: null;
+		const currentDate = new Date();
+		const currentTime = currentDate.getTime();
+		let timer = currentTime - calcDate;
+		if (item.isChecked && timer >= millisecondsIn24hrs) {
+			updateItem(listToken, item, {
+				isChecked: !item.isChecked,
+			});
+		}
+	}, []);
+
+	const handleChange = useCallback(async () => {
+		if (item.isChecked === false) {
+			await updateItem(listToken, item, {
+				isChecked: item.isChecked,
+				dateLastPurchased: new Date().getTime(),
+				totalPurchases: item.totalPurchases++,
```

Comment:
```suggestion
				totalPurchases: item.totalPurchases+1,
```

we have 2 problems here:
1. When the `++` operator is used after the operand it increases but returns the value before increasing so you do not receive the right value.
2. you mutate the object that you need to keep intact.